main.go

```diff
@@ -1000,16 +1000,48 @@ func downloadFromS3WithPrefix(client *minio.Client, prefix, bucket string) (stri
 	}
 	// if folder is included, strip it so it doesnt end up in a folder on the host itself
 	targetFilename := path.Base(filename)
+	targetFileLocation := fmt.Sprintf("%s/%s", backupBaseDir, targetFilename)
 	for retries := 0; retries <= s3ServerRetries; retries++ {
-		err := client.FGetObject(bucket, filename, backupBaseDir+"/"+targetFilename, minio.GetObjectOptions{})
+		object, err := client.GetObject(bucket, filename, minio.GetObjectOptions{})
 		if err != nil {
 			log.Infof("Failed to download etcd snapshot file [%s]: %v, retried %d times", filename, err, retries)
 			if retries >= s3ServerRetries {
 				log.Warningf("Failed to download etcd snapshot file [%s]: %v", filename, err)
 				break
 			}
 		}
+		defer object.Close()
+
+		localFile, err := os.Create(targetFileLocation)
+		if err != nil {
+			log.Warningf("Failed to create local file [%s]: %v", targetFileLocation, err)
+			if retries >= s3ServerRetries {
+				break
+			}
+		}
+		defer localFile.Close()
+
+		stat, err := object.Stat()
+		if err != nil {
+			log.Warningf("Failed to stat local file [%s]: %v", targetFileLocation, err)
+			if retries >= s3ServerRetries {
+				break
+			}
+		}
+
+		if _, err = io.CopyN(localFile, object, stat.Size); err != nil {
```

Comment:
i know the example (https://github.com/minio/minio-go/blob/master/examples/s3/getobject.go) shows using stat + copyN but the docs show just using io.Copy (https://docs.min.io/docs/golang-client-api-reference.html#GetObject) without the stat call.. maybe could just avoid the stat call?